### ISSUE TITLE
Python3.11

### ISSUE THREAD

**Issue by redigaffi** [emcache/emcache#79]
Any plans to upgrade to Python3.11?

**Comment by pfreixes** [comment]
happy to take any PR for providing compatibility with Python 3.11, but first we need to fix the pipeline .... @Hazealign any chance to have your for some time for fix the Ci, from latest PR merged we are still in a failure state unless Im missing something

**Comment by Hazealign** [comment]
I feel so sorry for everyone using emcache. Debugging GHA is not so easy. I'll try it in my forked repository and make PR ASAP.

**Comment by pfreixes** [comment]
no problemo @Hazealign , lets try to have this fixed

**Comment by Hazealign** [comment]
@pfreixes I made a patch (#81) for fix release pipeline. Can you review and test this?

**Comment by pfreixes** [comment]
This is no longer blocked, so any one interested can pick it up and provide support for Python 3.11, with new way of building weels should be relatively easy, any volunteer?

**Comment by Hazealign** [comment]
I investigated two issue to blocking to support Python 3.11. 

1. Cython version is too low for support Python 3.11. It's related with https://github.com/python/cpython/pull/28968, fixed with https://github.com/cython/cython/pull/4428 as v0.29.25. I check with latest Cython, and it compiles well.
2. [asynctest](https://github.com/Martiusweb/asynctest/) doesn't support Python 3.11. It's not maintained anymore. I think we should remove CoroutineMock in test, and test it manually.

I'm trying to fix tests now. When If I think I can't to do this. I'll add some comments.

**Comment by Hazealign** [comment]
@pfreixes 

I finally made working branch on [my forked repo](https://github.com/Hazealign/emcache). But I'm curious about those things.

1. Now we can't use `CoroutineMock` from `asynctest`, but `unittest.mock.AsyncMock` only supports >= Python 3.8. Should we discontinue Python 3.7 or try to find new alternative? (Python 3.7 is not EOL.)
2. Can we change Linux build image to `manylinux2010` to `manylinux2014`? `manylinux2010` is in EOL. So we can't build Python 3.11 targeted wheel file in `manylinux2010` docker image. (I already moved to manylinux2014 in working branch.)
